src/app/api/[[...route]]/routes/tenant.route.ts

```diff
@@ -14,12 +19,80 @@ app.delete('/:tenantId', (c) => {
   return c.json({ message: `Deleting tenant id: ${tenantId}` });
 });
 
-app.post('/:shareHouseId', (c) => {
-  const shareHouseId = c.req.param('shareHouseId');
-  return c.json({
-    message: `Creating share house id for tenant: ${shareHouseId}`,
-  });
-});
+app.post(
+  '/:shareHouseId',
+  zValidator('json', tenantInvitationSchema),
+  async (c) => {
+    try {
+      const shareHouseId = c.req.param('shareHouseId');
+      const data = c.req.valid('json');
+
+      const rotationAssignment = await prisma.rotationAssignment.findUnique({
+        where: {
+          shareHouseId: shareHouseId,
+        },
+        include: {
+          tenantPlaceholders: true,
+        },
+      });
+
+      if (!rotationAssignment)
+        return c.json(
+          { error: 'RotationAssignment not found for the given shareHouseId' },
+          404,
+        );
+
+      const availableTenantPlaceholder =
+        rotationAssignment.tenantPlaceholders.find(
+          (tenantPlaceholder) => tenantPlaceholder.tenantId === null,
+        );
+
+      let newTenant: Tenant;
+
+      if (availableTenantPlaceholder) {
+        newTenant = await prisma.tenant.create({
+          data: {
+            name: data.name,
+            email: data.email,
+            extraAssignedCount: 0,
+            tenantPlaceholders: {
+              connect: {
+                rotationAssignmentId_index: {
+                  rotationAssignmentId:
+                    availableTenantPlaceholder.rotationAssignmentId,
+                  index: availableTenantPlaceholder.index,
+                },
+              },
+            },
+          },
+        });
+      } else {
+        const lastIndex = rotationAssignment.tenantPlaceholders.length;
+        newTenant = await prisma.tenant.create({
+          data: {
+            name: data.name,
+            email: data.email,
+            extraAssignedCount: 0,
+            tenantPlaceholders: {
+              create: {
+                rotationAssignmentId: rotationAssignment.id,
+                index: lastIndex,
+              },
+            },
+          },
+        });
```

Comment:
Could you add a check whether there's already a tenant with the same email address because it has to be unique?

https://github.com/Tascurator/tascurator-documents/blob/main/docs/docs/modeling.md#tenant---%E3%83%86%E3%83%8A%E3%83%B3%E3%83%88

Reply:
I added the if statement.